src/ethereum/osaka/transactions.py

```diff
@@ -568,6 +584,40 @@ def validate_transaction(tx: Transaction) -> Tuple[Uint, Uint]:
     return intrinsic_gas, calldata_floor_gas_cost
 
 
+def count_access_list_bytes(access_list: Tuple[Access, ...]) -> Tuple[int, int]:
+    """
+    Count zero and non-zero bytes in access list addresses and storage keys.
+    
+    This function implements the byte counting mechanism from [EIP-7981], which
+    counts the actual zero and non-zero bytes in the raw address (20 bytes) and
+    storage key (32 bytes) data within the access list.
+    
+    Returns a tuple of (zero_bytes, nonzero_bytes).
+    
+    [EIP-7981]: https://eips.ethereum.org/EIPS/eip-7981
+    """
+    zero_bytes = 0
+    nonzero_bytes = 0
+    
+    for access in access_list:
+        # Count bytes in address (20 bytes)
+        for byte in access.account:
```

Comment:
Counting zero and non-zero bytes is being used in a couple of other places as well, including for call data in `calculate_intrinsic_cost`. Perhaps it would be a good idea to refactor it into a function in the `ethereum.utils.byte` module.